subprojects/core/src/main/java/org/gradle/api/internal/attributes/DefaultImmutableAttributesFactory.java

```diff
@@ -58,7 +62,15 @@ public AttributeContainerInternal mutable(AttributeContainerInternal parent) {
 
     @Override
     public <T> ImmutableAttributes concat(ImmutableAttributes node, Attribute<T> key, T value) {
-        return doConcatIsolatable(node, key, isolatableFactory.isolate(value));
+        return doConcatIsolatable(node, key, isolate(value));
+    }
+
+    private <T> Isolatable<T> isolate(T value) {
+        if (value instanceof String) {
+            return (Isolatable<T>) new CoercingStringValueSnapshot((String) value, instantiator);
```

Comment:
@jjohannes You might want to revert this on your branch, so that an explicit String attribute is incompatible with Named again. Or you might want to go the other way and make Named compatible with String both ways.

Reply:
Thanks. I will check my branch again once this PR is merged.